src/Analysis/Ast/Impl/Modules/Resolution/ModuleResolutionBase.cs

```diff
@@ -105,11 +103,24 @@ public IPythonModule GetImportedModule(string name)
         }
 
         protected void ReloadModulePaths(in IEnumerable<string> rootPaths) {
-            foreach (var moduleFile in rootPaths.Where(Directory.Exists).SelectMany(p => PathUtils.EnumerateFiles(_fs, p))) {
-                PathResolver.TryAddModulePath(moduleFile.FullName, moduleFile.Length, false, out _);
+            foreach (var root in rootPaths) {
+                foreach (var moduleFile in PathUtils.EnumerateFiles(_fs, root)) {
+                    PathResolver.TryAddModulePath(moduleFile.FullName, moduleFile.Length, false, out _);
+                }
+
+                if (PathUtils.TryGetZipFilePath(root, out var zipFilePath, out var _) && File.Exists(zipFilePath)) {
+                    foreach (var moduleFile in PathUtils.EnumerateZip(zipFilePath)) {
+                        if (!PathUtils.PathStartsWith(moduleFile.FullName, "EGG-INFO")) {
+                            PathResolver.TryAddModulePath(
+                                Path.Combine(zipFilePath,
+                                PathUtils.NormalizePath(moduleFile.FullName)),
+                                moduleFile.Length, false, out _
```

Comment:
So what would be the `fullModuleName` returned by the `TryAddModulePath`?

Reply:
`fullModuleName` returns the empty string, not sure if that's what is supposed to happen

Reply:
Is that an expected full module name?

Reply:
Sorry I should have clarified that a root was `test.zip/test`. I ran the test on the library `simplejson` and for files like `simplejson/decoder.py` it returns `simplejson.decoder` and for `simplejson/tests/test_bitsize_int` it gives `simplejson.tests.tesT_bitsize_int` etc, so I believe this is correct.

Reply:
Ok. Then let's add an import test to ensure that import of the module from egg is resolved correctly.

Reply:
Yes I did that in:
 https://github.com/microsoft/python-language-server/pull/1477/files#diff-fc769198933ce6a497889cd518874a9dR897-R928

It loads up the simplejson library and checks to see if some variables have members